vendor/github.com/ethereum/go-ethereum/les/fetcher.go

```diff
@@ -195,6 +196,10 @@ func (f *lightFetcher) syncLoop() {
 				f.pm.serverPool.adjustResponseTime(req.peer.poolEntry, time.Duration(mclock.Now()-req.sent), req.timeout)
 			}
 			f.lock.Lock()
+			if f.dbClosed {
```

Comment:
on line 133 there is a quitSync channel, which is closed in protocolManager and then protocol manager actually waits for all goroutines to finish with sync group before closing database

we need to find why it doesn't work as designed

Reply:
I thought it was failing because of this scenario:
1) `case resp := <-f.deliverChn:`
2) `f.processResponse()`
3) From other goroutine: `close(pm.quitSync)` and then later `chainDb.Close()`
4) `InsertHeaderChain` (will cause the error)
5) `case <-f.pm.quitSync: return` (too late)

If this doesn't work, I must be missing another place where `chainDb` is getting `Close`d too early.

Reply:
@dshulyak @canercidam  Can we try to add more logs for running on Travis?

Reply:
i made one run with higher verbosity for whisper tests https://travis-ci.org/status-im/status-go/jobs/344199616

Reply:
I can't see it getting closed anywhere else. So now, I'm trying to find out where another `InsertHeaderChain` call could be happening after `chainDb` is closed.

@mandrigin I want to but do I need to update the patch for vendor check?

Reply:
I think it's happening because of `eth/downloader/downloader.go`, L1272.